Collimation halo studies

mfiascar opened the pull request:
I have added some efficiency computations and output files for collimation halo studies

kyrsjo commented:
Hi, quick comment: I'l have to check the unit number for your output file (99) - I would pick something higher, that is usually safer. It should also be protected with an `inquire`, to see that nobody else has already taken whatever unit number we're using.

kyrsjo commented:
Example, opening (for reading):
```
         inquire( unit=664, opened=lopen )
         if (lopen) then
+if cr
            write(lout,*)"DYNK> **** ERROR in dynk_parseFUN():FILE ****"
            write(lout,*)"DYNK> unit 664 for file '"//
     &           cexpr_dynk(ncexpr_dynk), "' was already taken"
+ei
+if .not.cr
            write(*,*)   "DYNK> **** ERROR in dynk_parseFUN():FILE ****"
            write(*,*)   "DYNK> unit 664 for file '"//
     &           cexpr_dynk(ncexpr_dynk), "' was already taken"
+ei
            call prror(-1)
         end if
         
         open(unit=664,file=cexpr_dynk(ncexpr_dynk),action='read',
     &        iostat=stat,status="OLD")
```

For collimation, you can probably skip the `+if cr / +if .not.cr` part, since it's anyway not (currently) supported.

Thanks!

mfiascar commented:
I have modified the output file unti number and added the 'inquire'

kyrsjo commented:
Perfect, thanks!

One more thing: Alessio has a Twiki page for the collimation version. Here, he lists the unit numbers of all files used in the collimation version.
https://twiki.cern.ch/twiki/bin/viewauth/LHCAtHome/SixTrackCollimatVer
Could you add your new files to this page once this is merged?

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -28766,7 +28795,65 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
+	endif
+      open(unit=1992, file='efficiency_dpop.dat')
+!UPGRADE 4/11/2014
+      if(n_tot_absorbed.ne.0d0) then
```
n_tot_absorbed is an integer. Please compare it to "0", not "0d0".

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -28743,13 +28767,18 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
+      endif
+      open(unit=1991, file='efficiency.dat')
 !UPGRADE JANUARY 2005
       if(n_tot_absorbed.ne.0d0) then
```
n_tot_absorbed is an integer. Please compare it to "0", not "0d0".

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -28766,7 +28795,65 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
+	  call prror(-1)
+      endif
+      open(unit=1993, file='efficiency_2d.dat')
+      if(n_tot_absorbed.ne.0d0) then
```
Also here.

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -29578,7 +29665,11 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
       do i = 1, numeff
 !hr08   rsig(i) = dble(i)/2d0 - 0.5d0 + 6d0
-        rsig(i) = (dble(i)/2d0 - 0.5d0) + 6d0                            !hr08
+        rsig(i) = (dble(i)/2d0 - 0.5d0) + 5d0                           !hr08
```
I see you're modifying what I assume to be some histogram bins - have you cleared this with the rest of the collimation group?

mfiascar replied:
Yes, actually I have extended the range of the histogram, there is no impact on other people.

kyrsjo replied:
Ah, I see, you also changed `numeff` which is the size of the array. OK 👍

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -839,15 +840,29 @@
+      integer counted(npart,numeff)
       double precision neff(numeff),rsig(numeff)
-      common  /eff/ neff,rsig
+      common  /eff/ neff,rsig,counted
```
Array `counted` is not actually used anywhere. Please remove it.

kyrsjo commented on SixTrack/sixtrack.s:
```diff
@@ -33161,30 +33260,44 @@ C     Convert r(1), r(2) from U(0,1) -> rvec0 as Gaussian with cutoff mcut (#sig
      &/                                                                 &
-     &(tbetay(ie)*myemity0))                                            &
+     &(tbetay(ie)*myemity0)                                             &
      &).ge.rsig(ieff)) then
```
When writing the changelog / commit message when merging the pull, we should probably mention that this now includes the alpha? Since this may affect the results of other people who rely on counted_r and neff...

mfiascar replied:
This was in fact a bug and I discussed it with Roderik (in fact he spotted it). Roderik and I are the only ones doing these kind of studies, but of course you are right it should be mentioned in the commit. Where should I add the comment?

kyrsjo commented:
Hi Maria,

This looks mostly fine now. Could you please take a look at my comments and address them? Once this is done, we can merge. Sorry that I wasn't completely clear when we spoke in the corridor earlier today.

One more thing: Just before we merge, we want to update the date (5th line in the file). To do this while avoiding any merge conflicts (due to multiple people modifying the same line in the code), we will first merge the current master onto your branch, then do the modification, and finally merge back to master.

This how this is done:
1. In your local copy, add the upstream master. In the bottom SixTrack folder, there is a hidden subfolder .git. In this, *add* the following lines at the bottom of the file:
```
[remote "upstream"]
        url = git@github.com:SixTrack/SixTrack
        fetch = +refs/heads/*:refs/remotes/upstream/*
```

2. Fetch (download, but don't unpack anything yet) the upstream:
`git fetch upstream`

3. Merge the upstream master onto your branch:
`git checkout fccST` then `git merge upstream/master`

4. Assuming there are no conflicts, let's push your updated branch to it's remote (i.e. your fork on GitHub): `git push`

5. While we're at it, let's also update your master branch to the newest upstream/master. We've already done `git fetch upstream`, so we can skip that. The next steps are then:
`git checkout master` to go to your master branch, `git merge upstream/master` to update your master branch with the current upstream/master, then `git push` to put it on your fork in the repository.

kyrsjo commented:
Ah, regarding the last point (updating your master): I see you committed some stuff directly to it by mistake. Then you can't really just update your master branch, since it's history will be messed up (it won't be a fast forward commit).

In order to fix it, you need to do a `git reset` on that branch to completely undo those commits, then a "forced push" to your fork. If you want, I could come to your office and we can do it together...

mfiascar commented:
Hi Kyrre,

thanks for all the comments and the instructions.

Yes, I committed to the master branch instead of the fcc one by mistake. I will try it out and come to see you in case of problems.

cheers,
Maria


On 14 Jul 2016, at 11:41, Kyrre Ness Sjøbæk <notifications@github.com<mailto:notifications@github.com>> wrote:


Ah, regarding the last point (updating your master): I see you committed some stuff directly to it by mistake. Then you can't really just update your master branch, since it's history will be messed up (it won't be a fast forward commit).

In order to fix it, you need to do a git reset on that branch to completely undo those commits, then a "forced push" to your fork. If you want, I could come to your office and we can do it together...

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/SixTrack/SixTrack/pull/84#issuecomment-232617573>, or mute the thread<https://github.com/notifications/unsubscribe/AHJMDykxDvlCRM5rxznN9H_hmQa5UeoQks5qVgRZgaJpZM4JIDvf>.

kyrsjo commented:
OK, so now everything is ready! Two more things I need from you:
1. Update the date in the top of the file
2. Write (just as a comment here, I will copy it when merging) a commit message. Thanks!

Once I have these, I will merge your improvements. Thanks!

mfiascar commented:
The date has been updated.
For the commit message:
"Fixed radial amplitude definition used for halo studies to include also the alpha and angle"

kyrsjo commented:
Regarding the commit message: Nothing about the new output file, and the changes in the histogram which rsig is related to?

mfiascar commented:
Sorry, I wrote to fast and forgot there was no previous commit message. The full changelog should be:

Modifications for collimation halo studies:
- increased the range of the efficiency histogram vs. radial amplitude (file efficiency.dat), now ranging from 5-20.5 sigma;
- fixed radial amplitude definition used for halo studies to include also the alpha and angle
- added to output files: 
         * efficiency_dpop.dat: efficiency vs. Delta p/p to be used for off-momentum halo studies
         * efficiency_2d.dat: 2D efficiency vs. radial amplitude and Delta p/p

Thanks a lot!

kyrsjo commented:
You may now click to delete the branch (unless you plan to add more changes to it and have another PR).
